pkg/operator/sync_controllermanager_v311_00.go

```diff
@@ -206,8 +222,108 @@ func manageServiceCatalogControllerManagerDeployment_v311_00_to_latest(client ap
+				switch envVar.Name {
+				case httpProxyEnvVar, strings.ToLower(httpProxyEnvVar):
+					forceRollout = forceRollout || proxyConfig.Status.HTTPProxy != envVar.Value
+					if proxyConfig.Status.HTTPProxy != envVar.Value {
+						klog.Infof("httpproxy [%s] != [%s]; forceRollout %v", proxyConfig.Status.HTTPProxy, envVar.Value, forceRollout)
+					}
+				case httpsProxyEnvVar, strings.ToLower(httpsProxyEnvVar):
+					forceRollout = forceRollout || proxyConfig.Status.HTTPSProxy != envVar.Value
+					if proxyConfig.Status.HTTPSProxy != envVar.Value {
+						klog.Infof("httpsproxy [%s] != [%s]; forceRollout %v", proxyConfig.Status.HTTPSProxy, envVar.Value, forceRollout)
+					}
+				case noProxyEnvVar, strings.ToLower(noProxyEnvVar):
+					forceRollout = forceRollout || proxyConfig.Status.NoProxy != envVar.Value
+					if proxyConfig.Status.NoProxy != envVar.Value {
+						klog.Infof("noproxy [%s] != [%s]; forceRollout %v", proxyConfig.Status.NoProxy, envVar.Value, forceRollout)
+					}
+				default:
+					klog.Infof("None of the cases matched. forceRollout: %v", forceRollout)
+				}
+			}
+
+			// if we detected a change, forceRollout will be set, update
+			// environment
+			if forceRollout {
+				klog.Info("we have a proxyConfig and a daemonset, we detected a change in the env, updating required env")
+				// update the EnvVar
+				addProxyToEnvironment(required, proxyConfig)
+			}
+		}
+	} else if proxyConfig == nil {
+		// use required, do not add environment, or make it empty
+		existing, err := client.DaemonSets(required.Namespace).Get(required.Name, metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			// do nothing
+			foundDaemonSet = false
+		} else if err != nil {
+			foundDaemonSet = false
+			klog.Error("Problem getting the daemonset, returning")
+			return nil, false, err
+		} else if err == nil {
+			foundDaemonSet = true
+		}
+
+		// need to blank out EnvVar
+		if foundDaemonSet {
+			// see if there was a proxy that needs to get unset
+			for _, envVar := range existing.Spec.Template.Spec.Containers[0].Env {
+				switch envVar.Name {
+				case httpProxyEnvVar, strings.ToLower(httpProxyEnvVar),
+					httpsProxyEnvVar, strings.ToLower(httpsProxyEnvVar),
+					noProxyEnvVar, strings.ToLower(noProxyEnvVar):
+
+					forceRollout = forceRollout || len(envVar.Value) > 0
+					if len(envVar.Value) > 0 {
+						klog.Infof("%s [%s] != ['']; forceRollout %v", envVar.Name, envVar.Value, forceRollout)
+					}
+				default:
+					klog.Infof("None of the cases matched. forceRollout unchanged: %v", forceRollout)
+				}
+			}
+			// if we detected a change, forceRollout will be set, update
+			// environment
+			if forceRollout {
+				klog.Info("we have no proxyConfig but we do have a daemonset, we detected a change in the env, updating required env")
+				required.Spec.Template.Spec.Containers[0].Env = append(
+					required.Spec.Template.Spec.Containers[0].Env, []corev1.EnvVar{}...)
```

Comment:
I was basically forcing the environment to be there and for it to be EMPTY! I may be able to not do it and it will be okay. The `ds.yaml` does not have it defined, so `required` won't have anything there. I was unsure what happened what the `resourceapply.ApplyDaemonSet(` would do if the `DaemonSet` had an environment or not. 

So I set it to empty. If `resourceapply.ApplyDaemonSet(` will erase any environment that exists on the running `DaemonSet`.

Reply:
IIUC this is a no-op. Since `[]corev1.EnvVar{}` is an empty slice, this statement is equivalent to:
```go
required.Spec.Template.Spec.Containers[0].Env = required.Spec.Template.Spec.Containers[0].Env
```

I'm guessing what we're going for is, "use the static environment from `ds.yaml`, but nothing else." 

If we were using `existing` here, there _would_ be a need to remove the proxy envvars from the list, but since we're using `required`, which is unmarshaled from `ds.yaml`, I think it should be safe to remove this line entirely.

Reply:
Fixed by this commit https://github.com/openshift/cluster-svcat-controller-manager-operator/pull/42/commits/cac738ebf27518778732f641fa9dd19dd283a414